### DISCUSSION TITLE
ADC name change or expand with humidity/battery.

### DISCUSSION THREAD

**Discussion by Basti149** [arendst/Tasmota#22179] [Ideas]
I have an idea and that is to rename the words "ADC1Range". 
Or add two more in the IO selection.
1. ADC Battery
2. ADC Moisture

Could you also add it to Tasmota?
Or is there a possibility via Rules?
Since these are percentage values, the percentage information is missing. This is definitely a bug in Tasmota

Symbol „Von der Community überprüft“

![20240921_180655](https://github.com/user-attachments/assets/1247da68-ee58-45b9-b665-3ebc513f0611)

**Comment by sfromis** [comment]
What you show is definitely no bug or "missing info". When configuring for `ADC Range`, you are simply remapping numbers, it is up to you what you want the numbers to mean. And the rudimentary Tasmota front page is not even meant to be "the place" to check the numbers.

With ESP32, you could create a driver in the Berry language to pick up values, and supply an additional sensor entry with whatever you prefer of conversion, labeling and units.

Still, Tasmota does have a config for `ADC Voltage` which looks like half of what you're after.

**Comment by Basti149** [comment]
As I understood correctly, ADC range is one percent from 0-100 and the percent sign is missing. (This is a bug in my opinion).

And otherwise I'm looking for a more precise name for the Gui to make it easier to distinguish, e.g
ADC range1 "Moisture content" (0-100%)

ADC range2 "Battery" (as a percentage value 0-100% and not in voltage volts 0-3.3V)

Holen Sie sich Outlook für Android<https://aka.ms/ghei36>
________________________________

**Reply by sfromis** [reply]
No, `ADC Range` has nothing to do with a percentage, and adding a `%` would be a bug. It is just a number, with a linear remapping using the points you can give with the `AdcParam` command.

Also, as I just told you, the rudimentary Tasmota front page is not even meant to be "the place" to check the numbers; that would normally be in your backend smarthome setup.

The only thing you could get for a battery would be a voltage, not the complex calculations for even pretending to give an unreliable percentage. The voltage curve for typical lithium cells is pretty flat most of the way, before dropping quickly near the end.
